sklearn/mixture/tests/test_gmm.py

```diff
@@ -327,6 +328,32 @@ def test_1d_1component():
         assert_array_almost_equal(g.bic(X), g_full_bic)
 
 
+def assert_fit_predict_correct(model, X):
+
+    predictions_1 = model.fit(X).predict(X)
+    predictions_2 = model.fit_predict(X)
+
+    assert adjusted_rand_score(predictions_1, predictions_2) == 1.0
```

Comment:
Is the argument that since we are setting the random seed to be the same, that 
```
assert np.all(predictions_1 == predictions_2)
```
is sufficient?